cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -939,6 +944,25 @@ def parse_bucket_url(url):
     return {"bucket_name": bucket_name, "object_key": object_key, "object_name": object_name}
 
 
+def generate_s3_presigned_url(s3_uri, expiration=3600):
+    """Generate a presigned URL to share an S3 object.
+
+    :param s3_uri: s3 uri, e.g. s3://my.bucket/my.object
+    :param expiration: Time in seconds for the presigned URL to remain valid
+    :return: Presigned URL as string
+    """
+    match = re.match(r"s3://(.*?)/(.*)", s3_uri)
+    if match:
+        bucket_name = match.group(1)
+        object_key = match.group(2)
+    else:
+        raise URLError("Invalid s3 url: {0}".format(s3_uri))
+
+    return boto3.client("s3").generate_presigned_url(
```

Comment:
we could use singleton instance AWSApi.instance().s3.generate_presigned_url instead of boto3 client and add generate_presigned_url to [s3.py](https://github.com/aws/aws-parallelcluster/blob/e227d4d936b43d76987a4a70293b4a26694df682/cli/src/pcluster/aws/s3.py#L16)

Reply:
Done, thanks!